### DISCUSSION TITLE
Questions about Functionality

### DISCUSSION THREAD

**Discussion by Jonop23** [petrleocompel/hikaxpro_hacs#62] [Q&A] [answered]
Hi, I am looking for an alarm system classified as grade 2 in Spain (which allows it to be monitored by a central monitoring station) that can be self installed and integrated into Home Assistant so that I can also utilise it for automations and avoid the need for 2 sets of sensors, one for the alarm and one for other automations. The Hikvision AX Pro system seems to be the only one that ticks those boxes thanks to this integration. The only other grade 2 classified system I can find is the Ajax system but the integration into Home Assistant only seems to offer functionality when the system is armed and the system cannot be armed from Home Assistant.

Some questions about this integration please:

- Does the integration allow for creating automations based on sensor status even when the system is not armed e.g. non-contact of a door/window magnet sensor when the system isn't armed?
- Does the integration allow for reading of presence/motion detection from PIR sensors and cameras? This would be great as I can avoid fitting additional motion sensors for automations
- Does the integration allow for arming/disarming of the alarm system through Home Assistant?

Many thanks

**Comment by petrleocompel** [comment, marked as answer]
You can answer some of those yourself by looking on the screenshots.

> Does the integration allow for creating automations based on sensor status even when the system is not armed e.g. non-contact of a door/window magnet sensor when the system isn't armed?

Yes, **but** !! There is update timeout. Sync we are pulling updates from system. Default pull time is 30 seconds.

> Does the integration allow for reading of presence/motion detection from PIR sensors and cameras? This would be great as I can avoid fitting additional motion sensors for automations

Cant tell. Maybe. Never tested. But seems yes PIR sensors provide attribute "status" which changes to "trigger"

> Does the integration allow for arming/disarming of the alarm system through Home Assistant?
Yes! Main functionality.

**Reply by Jonop23** [reply]
Thank you! Really appreciate the info.

Just 1 follow up question... can the pull time be reduced? If so what is the minimum?

**Reply by petrleocompel** [reply]
It can but it also depends on the system. Minimum ? 1s, 2s ? You can overload the system if you reduce the pull interval to some really low time.

So if your pull time would be 30 seconds and somebody would open and close window in this 30 seconds interval you might not get that information. 

I am still looking in to possibilities if we could handle "events" it is currently tracked in issue #2

**Reply by petrleocompel** [reply]
I have to say I was also in hunt for some alarm system compatible with HomeAssistant.

So far this has some documentation and at least it is actually secure! I am able to use with "lan only" mode. 

Happy to end up with HikVision AX Pro.

**Reply by Jonop23** [reply]
Thanks for the further info. Yes I was considering 1 s or less in order to have a quick response in an automation like turning on a light and a fan when someone opens a door or their presence is detected for example. Is it something you can please test to see if it does cause an overload?

I'm glad to hear you are happy with the AX Pro system! I have been researching for a while

**Reply by Jonop23** [reply]
Also, please can you help me fully understand the effect of the pull time with an example.. if someone opens a door and a magnet contact breaks. is the status change signal instantly pushed to or pulled by Home Assistant? Or do we have to wait for the status to be pulled from AX Pro by Home Assistant at the next defined interval?

**Reply by petrleocompel** [reply]
Minimal pull time is in seconds sooo 1 second. 
The home assistant each that time connects to the system to get status data. 

If there is too many requests the device can go offline... or not respond on the requests. If you would set for example 5 - 10 seconds it could be fine.

As I explained the "change" is not pushed to HA. That is something I am trying to make it work as described later in issue #2 .
HA connects to AX pro. That is the reason why it is named "pull".

**Reply by Jonop23** [reply]
Thanks for clarifying. Much appreciated!

**Comment by Esinem** [comment]
"Does the integration allow for reading of presence/motion detection from PIR sensors"  I'm really interested in this were to be possible when the alarm isn't armed, e.g trigger lights or even check if somebody is home. My sensors are wireless, BTW.

I'd rashly assumed it would be basic functionality, so wasted hours trying to figure it out as a complete newbie to HA. Wished I'd looked here first :-)  

I found this work around but no idea how it worked out: https://community.home-assistant.io/t/hikvision-ax-pro-alarm/541882/4 Thought I'd ask before opening another can of worms :-D

**Reply by petrleocompel** [reply]
It is possible, not really recommended.  Heavy CPU load on AXPro. It also depends on size of system.
Easier will be to point you to read the discussions about this

https://github.com/petrleocompel/hikaxpro_hacs/issues/143#issuecomment-2495888020
https://github.com/petrleocompel/hikaxpro_hacs/issues/2

**Comment by Esinem** [comment]
Thank you for your fast reply. I'm  not very techical, so I ony get a sense of the issue. Is this specific to AxPro or PIRs in general? Just wondering the best way to set up motion triggers Sent from Outlook for AndroidFrom: Petr Leo Compel ***@***.***>Sent: Tuesday, September 2, 2025 8:00:48 amTo: petrleocompel/hikaxpro_hacs ***@***.***>Cc: Esinem ***@***.***>; Comment ***@***.***>Subject: Re: [petrleocompel/hikaxpro_hacs] Questions about Functionality (Discussion #62)It is possible, heavy CPU load on AXPro.
Easier will be to point you to read the discussions about this#143 (comment)
#2—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.
